llarp/link/link_manager.cpp

```diff
@@ -1062,14 +1062,178 @@ namespace llarp
   void
   LinkManager::handle_path_build(oxen::quic::message m)
   {
+    if (!router.path_context().AllowingTransit())
+    {
+      log::warning("got path build request when not permitting transit");
+      m.respond(serialize_response({{"STATUS", PathBuildMessage::NO_TRANSIT}}), true);
+      return;
+    }
     try
     {
-      oxenc::bt_dict_consumer btdc{m.body()};
+      // not using list_consumer here because we need to move the first (our) frame
+      // to the end and re-send, unless we're the terminal hop.  This could be done
+      // with list_consumer, but it would involve messy mucking around with the
+      // encoded list.  Revisit if optimization issue (shouldn't be).
+      auto frame_list = oxenc::bt_deserialize<oxenc::bt_list>(m.body());
+      if (frames.size() != path::MAX_LEN)
+      {
+        log::info(
+            link_cat,
+            "Path build request has invalid number of records, ",
+            frame_list.size(),
+            "!=",
+            path::MAX_LEN);
+        m.respond(serialize_response({{"STATUS", PathBuildMessage::BAD_FRAMES}}), true);
+        return;
+      }
+
+      const auto& hash = frame_list[0].at("hash");
```

Comment:
`Type 'std::list<oxenc::bt_value>' does not provide a subscript operator` at least on my editor?